src/Bitmaps/BitmapFile.cpp

```diff
@@ -32,3 +32,14 @@ void BitmapFile::VerifyIndexedImageForSerialization(uint16_t bitCount)
 		throw std::runtime_error("Unable to read/write a non-indexed bitmap file. Bit count is " + std::to_string(bitCount) + " but must be 8 or less");
 	}
 }
+
+bool operator==(const BitmapFile& lhs, const BitmapFile& rhs) {
+	return lhs.bmpHeader == rhs.bmpHeader && 
+		lhs.imageHeader == rhs.imageHeader && 
+		lhs.palette == rhs.palette &&
+		lhs.pixels == rhs.pixels;
+}
```

Comment:
After the stuff I watched a little while ago on code formatting, I think I might recommend:
```cpp
return
    lhs.bmpHeader == rhs.bmpHeader && 
    lhs.imageHeader == rhs.imageHeader && 
    lhs.palette == rhs.palette &&
    lhs.pixels == rhs.pixels;
```

Related, should the `&&` go at the end of the line above, or the beginning of the line below? I was tempted to move them to the beginnings of the next lines, though I'm not sure that's a clear advantage.